### ISSUE TITLE
Optional "limited respawns" in config

### ISSUE THREAD

**Issue by Jamesthe1** [KosmosisDire/TeammateRevive#58]
**Is your feature request related to a problem? If so, please describe.**
I've noticed that teammate revives are, while handy, a bit strong for the game. Death has less been a worry and moreso an annoyance, but at the same time, a single death can also be a pain.

**Describe the feature you'd like**
To solve the above, I suggest limited respawns for those who want them. It'd still allow revives and the like while, at the same time, allowing for moments like having the last survivor standing.

**Describe alternatives you've considered**
I've considered changes to the default settings and the death curse, though there are some problems. Team revive becoming slower would make it more difficult, but in the end there's still that lack of threat. With death curse, not reviving a teammate for strategic reasons ends up making them feel left out more than if the respawn mechanic didn't exist.

**Comment by amadare42** [comment]
Hey, @Jamesthe1 thanks for the suggestion. 

After a lot of time spent on trying to balance Death Curse mod, I'm pretty sure that it is impossible to add limitless reviving that will not make game easier. The reason I personally and I assume a lot of other player will have this mod is to minimize gameplay downtime. When you're showing RoR2 to your friend in coop, you will inevitably have skill gap that will lead to your friend dying and be forced to just watch as you play. That isn't very fun.

And I think, this is important not only in high skill-gap situation. Original game approach have problem that if character dies, they have less time to gather items and will most probably loose access to level boss items. That lead to situation that other characters will become more powerful. That is somewhat mitigated if you do loop or two, but if you want to fight Mythrix loopless, it is feels bad even if you fine with having some gameplay downtime.

For these reasons, no matter what approaches I experimented, I never wanted to sacrifice limitless revives. I just tried to make dying more impactful with longer-lasted consequences that are felt immediately. And I still feel that way - I think limitless revives just make game more enjoyable for all characters even if making it somewhat easier.

I suggest approaching reviving as necessity. Like you always need to do it. Death Curse designed in a way that will cut HP a lot on first curse, but it is less noticeable with each consequent curse, so if your team dying a lot, you as a group can just decide to be okay with having curse or two. And unlike real Shaped Glass buff, it will not remove one-hit protection, so if you're good, you can survive with 10+ curses and even receive some shots from time to time.

You can also enforce this and add extra challenge by buying Dead Man's Hand with Newt. It will ensure no one will be left out. :)
![image](https://user-images.githubusercontent.com/3945692/180625305-5ab1e709-3e6e-419f-b565-8808e20c1eb8.png)

So in summary, I acknowledge issue you're trying to solve here - there is one. But I feel that having limited revives will just make game less enjoyable. Curious what @nathanpovo thinks.
